### DISCUSSION TITLE
Scroll to the top automatically in alpine js

### DISCUSSION THREAD

**Discussion by Shadecute** [alpinejs/alpine#2647] [1. Help] [answered]
hi,
i'm using this code for scroll to top automatically
```
<script>window.scrollTo({top: 0, behavior: 'smooth'})</script>
```
Is there an alternative to this code in Alpine js?

**Comment by chrisdevelops** [comment, marked as answer]
```
# Automatically scrolls to top upon loading
<div 
  x-data
  x-init="window.scrollTo({top: 0, behavior: 'smooth'})"
  class=""></div>

# Scrolls to top of the page on click
<button 
  x-data
  @click.window="window.scrollTo({top: 0, behavior: 'smooth'})"
  class="">Top</button>
```

**Reply by arewtech** [reply]
this work, tq

**Reply by grobiwebdesign** [reply]
This is a good answer, but if you want to use the "on click" method, it's better to write "@click="window.scrollTo" as this will confine the function to the button, otherwise the whole page scrolls back to the top clicking outside the button.
